src/main/java/org/sonarsource/sonarlint/ls/ServerMain.java

```diff
@@ -98,4 +101,14 @@ public static void main(String... args) {
     System.exit(exitCode);
   }
 
+  public static class MavenVersionProvider implements IVersionProvider {
```

Comment:
How is it related to Maven?

Reply:
My original version in #402 provided that name because it was taken from the Maven attributes in JAR's manifest. Probably, renaming is a good idea.

```suggestion
  public static class SonarlintVersionProvider implements IVersionProvider {
```

Reply:
It is related to Maven because `slls-version.txt` is a Maven-filtered resource that contains the `${project.version}` property.

I believe it is slightly more robust than the initial implementation because:
* This file is provided only by the language server build itself - it is not automagically generated by some Maven plugin or copied from a random JAR when building the über-JAR with dependencies
* It contains exactly the property that we want to show with the `--version` option